SwiftLeeds/Push/AppDelegate+Push.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  AppDelegate+Push.swift
+//  SwiftLeeds
+//
+//  Created by Matthew Gallagher on 04/09/2022.
+//
+
+import UIKit
+import UserNotifications
+
+extension AppDelegate {
+    func requestPushAuthorization(application: UIApplication) {
+        let notificatioNCenter = UNUserNotificationCenter.current()
+        notificatioNCenter.requestAuthorization(options: [.badge, .sound, .alert]) { [weak self] isGranted, error in
+            guard isGranted else { print("⛔️ not granted"); return }
+            if let error = error { print("⛔️", error); return }
+
+            notificatioNCenter.delegate = self
+
+            DispatchQueue.main.async {
+                application.registerForRemoteNotifications()
+            }
+        }
+    }
+
+    func sendPushRegistrationDatails(to url: URL, deviceToken: Data) {
+        var details = TokenDetails(token: deviceToken)
+
+#if DEBUG
+        details.debug = true
+        print("🚀", details)
+#endif
+
+        var request = URLRequest(url: url)
```

Comment:
It would be best to use the existing network stack instead of ignoring it and going for raw URL sessions if possible? I can't see any reason why not to use it in this instance

It would also be great to make sure we can try catch this request too so when debugging we could tell if the requests failed or not, rather than having them just fail silently like we have here ( using the network stack would help with this, but failing that just a standard try catch around the async version of these APIS would be great )

Reply:
This wasn't possible with the existing Network layer due to it not being written to handle body elements required for this call.  Therefore I used standard networking to handle it and will look at putting in a more production ready network stack in the future, but didn't want to hold up this work due to wanting to release the App soon.  

I can add some additional error checking around the actual call.

Reply:
There's a branch [here](https://github.com/SwiftLeeds/swiftleeds-ios/compare/patch/push-endpoint?expand=1) with a quick demo of changing it over

I noticed the API is throwing a bunch of `422` or `500` which was caught with the try catch, im not sure if there's a bug with the web code or not? `422` is probably expected if we get duplicates of the same token but `500` im un-sure, I'll leave that with you

Reply:
Yeah, I noticed this as well @SwiftyAlex it's reporting some critical errors within Heroku 🤔

Reply:
Already resolved, just needs more testing. It was not updating if already there